MediaAdmin/FileUtils/Image/ImagickImageManager.php

```diff
@@ -57,10 +57,9 @@ protected function resizeImage(array $format, Imagick $image)
             $image->setimagebackgroundcolor('#000000');
             $refRatio = $maxWidth / $maxHeight;
             $imageRatio = $image->getImageWidth() / $image->getImageHeight();
-
-            if ($refRatio > $imageRatio || $maxWidth == -1) {
+            if ($imageRatio >= 1 / $maxHeight && ($refRatio > $imageRatio || $maxWidth == -1)) {
                 $image = $this->resizeOnHeight($image, $maxHeight);
-            } else {
+            } else if ($imageRatio >= 1 / $maxWidth) {
```

Comment:
I think there's a problem with this refacto : it seems you can have cases where no image is generated.
There must always be a generated image, if a size (width or height) is a problem because it's too short, the fix was to artificially increase it.